casual/casual-caller/src/main/java/se/laz/casual/connection/caller/FailoverAlgorithm.java

```diff
@@ -114,6 +131,64 @@ private <T> T issueCall(String serviceName, List<ConnectionFactoryEntry> validEn
         throw new CasualResourceException("Call failed to all " + validEntries.size() + " available casual connections.", thrownException);
     }
 
+    /**
+     * Try to use a stickied pool if pool stickiness is configured and any stickied pool is available and serves the requested service
+     *
+     * @param serviceName Service to call
+     * @param factories   Currently available factories for service to call
+     * @param doCall      Provided service call procedure
+     * @return Optional ServiceReturn. An empty result could indicate that stickiness isn't enabled, sticky isn't set yet for the current transaction (which will be the case for the first call) or the stickied pool was unavailable so call to sticky was skipped. Empty should always lead to retry down the line if possible, otherwise a TPENOENT response.
+     * @throws ResourceException Some softer errors are reported as resource exceptions. If these are thrown later retries with other pools is possible.
+     */
+    private <T> Optional<T> handleTransactionSticky(
+            String serviceName,
+            List<ConnectionFactoryEntry> factories,
+            FunctionThrowsResourceException<CasualConnection, T> doCall) throws ResourceException
+    {
+        if (TransactionPoolMapper.getInstance().isPoolMappingActive())
+        {
```

Comment:
prefer shortcut returns on negation to reduce large if blocks.
e.g.  
```
if( ! TransactionPoolMapper.getInstance().isPoolMappingActive() )
{
    return Optional.empty();
} 
```

Reply:
You are correct, that makes for a nicer method. Deep nesting and very wide code blocks get nasty fast